tests_metricflow/fixtures/connection_url.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import urllib.parse
+from dataclasses import dataclass
+from typing import Optional, Sequence, Tuple
+
+
+@dataclass(frozen=True)
+class UrlQueryField:
+    """Field name / values specified in the query part of a URL."""
+
+    field_name: str
+    values: Tuple[str, ...]
+
+
+@dataclass(frozen=True)
+class SqlEngineConnectionParameterSet:
+    """Describes how to connect to a SQL engine."""
+
+    url_str: str
+    dialect: str
+    query_fields: Tuple[UrlQueryField, ...]
+    driver: Optional[str]
+    username: Optional[str]
+    password: Optional[str]
+    hostname: Optional[str]
+    port: Optional[int]
+    database: Optional[str]
+
+    # Custom-handling for Databricks.
+    http_path: Optional[str]
+
+    @staticmethod
+    def create_from_url(url_str: str) -> SqlEngineConnectionParameterSet:
+        """The URL roughly follows the format used by SQLAlchemy.
+
+        * This implementation is used to avoid having to specify SQLAlchemy as a dependency.
+        * Databricks has a URL with a semicolon that separates an additional parameter. e.g.
+          `databricks://host:port/database;http_path=a/b/c`. Need additional context for why this is done.
```

Comment:
Our original Databricks client was built before they added an officially supported SQLAlchemy client, so we used the JDBC connection URI. https://docs.databricks.com/en/integrations/jdbc/authentication.html

Reply:
Added in comment.